synapse/lib/ast.py

```diff
@@ -3239,13 +3239,20 @@ async def getPropAndValu(self, runt, path):
         if not path:
             return None, None
 
-        name = await self.kids[0].compute(runt, path)
+        propname = await self.kids[0].compute(runt, path)
+        name = await tostr(propname)
 
         ispiv = name.find('::') != -1
         if not ispiv:
 
             prop = path.node.form.props.get(name)
             if prop is None:
+                if not isinstance(propname, str):
+                    styp = await s_stormtypes.totype(propname, basetypes=True)
+                    mesg = f"Property names must be stringlike, got '{styp}'."
+                    err = s_exc.StormRuntimeError(mesg=mesg, name=propname, type=styp)
```

Comment:
Do we want to include the value in the message as well?
An Optic user would not be seen the `name` or `type` properties.

Reply:
updated to include the value